cmd/access/node_builder/access_node_builder.go

```diff
@@ -71,17 +71,16 @@ type AccessNodeBuilder interface {
 	IsStaked() bool
 }
 
-// AccessNodeConfig defines all the user defined parameters required to bootstrap an access node
+// SharedNodeConfig defines all the user defined parameters required to bootstrap an access or observer node
 // For a node running as a standalone process, the config fields will be populated from the command line params,
 // while for a node running as a library, the config fields are expected to be initialized by the caller.
-type AccessNodeConfig struct {
-	staked                       bool
+type SharedNodeConfig struct {
```

Comment:
nit: now that the observer and access node configs are split, you don't really need multiple struct changes. the original config was already preconfigured for access nodes

Reply:
Fair enough. It may still improve visibility in case of some common scenarios, but I will remove them for now.